### ISSUE TITLE
Update Module to Support GKE

### ISSUE THREAD

**Issue by kevensen** [forseti-security/terraform-google-forseti#169]
In order to support deploying forseti-security on GKE, this module should deploy a GKE cluster.  It should have the following features.

1. Deploy forseti-security via Helm
2. If the user already has a GKE cluster they want to use, don't deploy a new GKE cluster, and don't deploy a server VM
3. If the user doesn't have a GKE cluster, deploy a GKE cluster with hardening best practices, and don't deploy a server VM
4. If the user doesn't want to deploy on GKE, deploy the server VM

 - [x] Terra-form sub-module - For Helm to function on GKE (BETA)
 - [x] Terra-form changes for optional deployment of Server VM (GA)
 - [ ] Terra-form changes for optional deployment of Client VM (GA)

**Comment by morgante** [comment]
If we pursue this, it should be done as a submodule.

However, I'm not quite sure what value does having Terraform orchestrate the creation provide over simply deploying the Helm chart

**Comment by kevensen** [comment]
After further consideration, it may be enough to provide an alternative "main.tf".  Saying if you want to deploy this on gke, use the [GKE CFT module](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine).  Then add the [Helm Release](https://www.terraform.io/docs/providers/helm/release.html) resource to your main.tf

**Comment by RexBelli** [comment]
I've been talking with @kevensen, I'd like to take this on.  If someone can assign this issue to me, that'd be great.

**Comment by kevensen** [comment]
@morgante Terraform would create the GKE cluster.  The Helm chart will deploy forseti-security on GKE.

**Comment by morgante** [comment]
Definitely, if you're deploying to GKE you don't need a server VM. I can see the value in an alternative submodule which doesn't include that but does include the Helm Release.

**Comment by kevensen** [comment]
Gotcha.  I need to understand submodules better.

**Comment by aaron-lane** [comment]
@kevensen They're also referred to as "nested modules" in the [Standard Module Structure article](https://www.terraform.io/docs/modules/index.html#standard-module-structure). Their behaviour is completely arbitrary, but they're generally used provide composable units of related resources or alternative implementations of the root module.

**Comment by kevensen** [comment]
@aaron-lane Thanks for the info!  I'll look into it.

**Comment by kevensen** [comment]
The approach we've chosen is to place the necessary module invocations in a single main.tf.  Please let me know if there better practice.

**Comment by morgante** [comment]
You could create a PR to provide a nested module, we certainly wouldn't require users to create a local sub-module.

**Comment by kevensen** [comment]
@morgante  I am still learning Terraform best practices so I appreciate your guidance!  @RexBel, do you think we could refactor the work you've done to be a submodule?

**Comment by RexBelli** [comment]
@kevensen I originally attempted to do this and ran into an issue: I could not get the GKE cluster to be conditional.

My approach was to use the `count` parameter on any resource that needed to be conditional.  I found [Terraform modules do not support this parameter](https://github.com/hashicorp/terraform/issues/953), so I then moved to adding it to individual resources.  In this context the big ones would be the [GCE VMs in the `server` submodule](https://github.com/forseti-security/terraform-google-forseti/blob/master/modules/server/main.tf#L420), and the GKE cluster in the new `gke` submodule I was building.  This seemed like it was going to work until I ran into [this](https://github.com/hashicorp/terraform/issues/21557): I could not get the `count` parameter to work on the `google_container_cluster` resource.

I don't know enough about the inner workings of Terraform to say whether the issue is a bug or user error.
